HACKING-URL.md

```diff
@@ -0,0 +1,214 @@
+
+Introduction
+============
+
+This file contains instructions on how to deal with feature requests for
+supporting new URL domains in MusicBrainz, or to change how they are currently
+supported.
+
+Our URL support has three main parts: cleanup, favicons and sidebar display.
+Often you will want to add all three, but in some cases you might only want
+to display the links more prominently without cleaning them up, or vice versa.
```

Comment:
* On “cleanup”: The current name “URLCleanup” of the file is misleading: It is more broad than that as you know and described in the next section. I suggest to use “editing handler” (or any other suggestion?) instead here.
* On “favicons and sidebar display”: The pretty name and the scheme override are missing from this list. Those could all be grouped under the same part “external links display”.

Reply:
The suggestion for “editing handler” didn’t come out of nowhere. The term “handle” has been massively used about URLs/external links in both [pull requests](https://github.com/metabrainz/musicbrainz-server/pulls?q=is%3Apr+handle) and [tickets](https://tickets.metabrainz.org/issues/?jql=project%20%3D%20MBS%20AND%20component%20in%20(%22URL%20cleanup%22)%20AND%20text%20~%20handle) for a long time.

Reply:
Sure :) You had said back then you were going to make a bunch of changes, that's why I didn't touch this further :) But I can change it myself if you prefer.